Update gesture action selection UI

jpelgrom opened the pull request:
<!--
    Please review the contributing guide before submitting: https://developers.home-assistant.io/docs/android/submit
    Please, complete the following sections to help the processing and review of your changes.
    Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).

    Thank you for submitting a Pull Request and helping to improve Home Assistant. You are amazing!
-->

## Summary
<!--
    Provide a brief summary of the changes you have made and most importantly what they aim to achieve.
    Don't forget any links that could be useful to the reader. (Github issues, PRs, documentation, articles, ...)

    * What was the motivation behind this change?
    * What is the impact of the changes on the application?
-->
This PR updates the UI for selecting a gesture action from a dropdown to a full screen, grouping actions by a category.

Based on [design input](https://discord.com/channels/330944238910963714/1346948551892009101/1395373488512176158), while using existing components as much as possible. These changes are done for #5538 which will add a lot of actions, but I'm submitting this separately for easier review as it contains a new pattern with Compose Navigation code in a fragment. 

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [ ] New or updated tests have been added to cover the changes following the testing [guidelines](https://developers.home-assistant.io/docs/android/testing/introduction).
  _Adding Compose Navigation tests here feels out of scope as it requires a general Compose testing setup/this is not a navigation framework for the entire app._
- [x] The code follows the project's [code style](https://developers.home-assistant.io/docs/android/codestyle) and [best_practices](https://developers.home-assistant.io/docs/android/best_practices).
- [x] The changes have been thoroughly tested, and edge cases have been considered.
- [x] Changes are backward compatible whenever feasible. Any breaking changes are documented in the changelog for users and/or in the code for developers depending on the relevance.

## Screenshots
<!--
    If this is a user-facing change not in the frontend, please include screenshots in light and dark mode.

    Note: Remove this section if there are no screenshots.
-->
|Light|Dark|
|---|---|
|![A screen titled 'Swipe up using three fingers', showing radio buttons and headings, light mode](https://github.com/user-attachments/assets/d9221ed0-c3f7-4a4d-b863-2bd8e5c4e9bb)|![A screen titled 'Swipe up using three fingers', showing radio buttons and headings, light mode](https://github.com/user-attachments/assets/26d53b5f-ee7b-49a1-bd81-c1c0c100a25b)|

## Link to pull request in documentation repositories
n/a

## Any other notes
<!-- 
    If there is any other information of note, like if this Pull Request is part of a bigger change, please include it here.
-->

jpelgrom commented on common/src/main/res/values/strings.xml:
```diff
@@ -298,11 +298,20 @@
     <string name="gestures_direction_left">Swipe left</string>
     <string name="gestures_direction_right">Swipe right</string>
     <string name="gestures_direction_up">Swipe up</string>
+    <string name="gesture_name_swipe_down_two">Swipe down using two fingers</string>
+    <string name="gesture_name_swipe_down_three">Swipe down using three fingers</string>
+    <string name="gesture_name_swipe_left_two">Swipe left using two fingers</string>
+    <string name="gesture_name_swipe_left_three">Swipe left using three fingers</string>
+    <string name="gesture_name_swipe_right_two">Swipe right using two fingers</string>
+    <string name="gesture_name_swipe_right_three">Swipe right using three fingers</string>
+    <string name="gesture_name_swipe_up_two">Swipe up using two fingers</string>
+    <string name="gesture_name_swipe_up_three">Swipe up using three fingers</string>
```
These strings are new because simply using `direction + " " + number of pointers` may not work for all languages.

TimoPtr commented on app/src/main/kotlin/io/homeassistant/companion/android/settings/gestures/views/GestureActionsView.kt:
```diff
@@ -0,0 +1,96 @@
+        )
+        Text(
+            text = stringResource(action.description),
+            style = MaterialTheme.typography.body1.merge(),
```
Why merge?

TimoPtr replied:
For me it does nothing if we don't give it a parameter right?

TimoPtr commented on app/src/main/kotlin/io/homeassistant/companion/android/settings/gestures/views/GestureActionsView.kt:
```diff
@@ -0,0 +1,96 @@
+        Text(
+            text = stringResource(action.description),
+            style = MaterialTheme.typography.body1.merge(),
+            modifier = Modifier.padding(start = 32.dp),
```
Why 32dp here? Shouldn't we have the same padding on the left and right of the RadioButton?

jpelgrom replied:
To align the text with the title in the toolbar, otherwise you have text at three different positions from the start which looks messy.

jpelgrom commented:
> I think it misses one vertical padding around the list

Fixed:

|Scrolled to top|Scrolled to bottom|
|---|---|
|<img width="1080" height="1200" alt="image" src="https://github.com/user-attachments/assets/622eb37e-eac0-487c-b1c1-888bd8805aac" />|<img width="1080" height="1200" alt="image" src="https://github.com/user-attachments/assets/747045d7-a827-4a3b-8bdd-513668882c71" />|

> > Navigation is kinda weird, when I click on a specific gesture and I press back in the topbar
> 
> Hadn't found that one in testing, good catch!

Fixed: the settings activity actually had a weird setup where it would check the SupportFragmentManager backstack first, only then the back handler. Changed to always use back handler as those will always match.

TimoPtr commented:
> @TimoPtr Good to merge this now then?

Yes I approved with giving you the choice to make the screenshots here or in the next PR 😊
